test/playwright/e2e/filters.spec.ts

```diff
@@ -0,0 +1,225 @@
+import { test, expect, Page } from '@playwright/test'
+
+import {
+  assertCheckboxStatus,
+  openFilters,
+  changeContentType,
+  goToSearchTerm,
+  enableNewHeader,
+  closeFilters,
+  isPageDesktop,
+} from '~~/test/playwright/utils/navigation'
+
+import { mockProviderApis } from '~~/test/playwright/utils/route'
+
+import { TestScreen, testScreens } from '~~/test/playwright/utils/breakpoints'
+
+import { SCREEN_SIZES } from '~~/src/constants/screens'
+
+import {
+  supportedSearchTypes,
+  ALL_MEDIA,
+  IMAGE,
+  AUDIO,
+} from '~/constants/media'
+
+test.describe.configure({ mode: 'parallel' })
+
+const assertCheckboxCount = async (
+  page: Page,
+  checked: 'checked' | 'notChecked' | 'total',
+  count: number
+) => {
+  const checkedString = {
+    checked: ':checked',
+    notChecked: ':not(:checked)',
+    total: '',
+  }[checked]
+  const locatorString = `input[type="checkbox"]${checkedString}`
+  await expect(page.locator(locatorString)).toHaveCount(count, { timeout: 200 })
+}
+
+const FILTER_COUNTS = {
+  [ALL_MEDIA]: 11,
+  [AUDIO]: 32,
+  [IMAGE]: 70,
+}
+
+for (const breakpoint of testScreens) {
+  test.describe(`filters on ${breakpoint}`, () => {
+    const width = SCREEN_SIZES.get(breakpoint as TestScreen) as number
+
+    test.use({ viewport: { width, height: 700 } })
+
+    test.beforeEach(async ({ context, page }) => {
+      await mockProviderApis(context)
+      await enableNewHeader(page)
+    })
```

Comment:
The main reason was _I didn't understand how `describeBreakpoints` utility works_ 🤦‍♀️ I thought it was only for the snapshots, because that's how it's used in the VR tests, and after this comment I realized that I don't have to use the `expectSnapshot` that it provides :) I'm going to re-factor the tests to use it.
I think for the functionality tests, we can simply test one mobile and one desktop breakpoint. This will make sure that the functionality is tested, but will not increase the testing time too much (as if we tested all breakpoints).

Reply:
Right on! If you use `describeEachBreakpoint(testScreens)` you'd get the behaviour you're looking for (only `sm` and `xl` tested). Agreed that for non-snapshot tests we can skip all the intermediary sizes and save some time :racing_car:

Reply:
I've added a new alias `describeMobileAndDesktop`, and added documentation comments stating that it's to be used with e2e tests. It's easier for me to understand than `endToEndTests` that it has 2 variants from this name, even if that can be ambivalent since the `describeEvery` also basically tests mobile and desktop breakpoints.

Reply:
Nice! Despite the ambiguity I think it's a great name :grin: